### DISCUSSION TITLE
Removal of download.checksum.sha-512 from artifact.xml file of Eclipse P2 repo generated through Tycho

### DISCUSSION THREAD

**Discussion by ganesh08** [eclipse-tycho/tycho#2506] [Q&A]
I am using Maven Tycho for building Eclipse plug-ins(Tycho pomless version 3.0.3)

I would like to remove the download.checksum.sha-512 from artifacts.xml present inside artifacts.jar of P2 repository
Can you please suggest how to stop generating SHA-512 in artifact.xml file ?

Example sha-512 entry in artifact.xml  <property name='download.checksum.sha-512' value='xxxxxxxxxxxxxx'>

**Comment by laeubi** [comment]
This is actually added by P2 an not really configurable, the most interesting question would be why you want to remove it at all?

**Reply by ganesh08** [reply]
The generated Eclipse P2 repose using tycho  is not able to consume as eclipse p2 target . When this repo is loaded in eclipse pde target getting error Sha512 error hence want to generate P2 repos without Sha512.
From eclipse version 2022-12 Eclipse if SHA 512 entry present not able to load it as target

**Reply by laeubi** [reply]
Can you please share a reproducer? Additional checksums should not prevent the target from being loaded, this looks like a bug in P2/PDE ...

**Reply by akurtakov** [reply]
Furthermore from p2 side I would rather see older checksums being faded out (for security reasons) than making configurable removing the strongest one we have.

**Reply by ganesh08** [reply]
Can be reproduced by using below example
https://github.com/klibio/example.pde.rcp -> Checkout build and check file, releng\example.rcp.site\target\repository\artifacts.jar
Load the garnered P2 repos in eclipse as target

**Comment by laeubi** [comment]
I have downloaded your example and added the result to a target file (using latest eclipse release 2023-03):

![grafik](https://github.com/eclipse-tycho/tycho/assets/1331477/88d6a6f7-8a22-4a5a-ad57-a23c5719fbfe)

so I can't see any immediate issue with that update site in general maybe you can share a complete error message?
